pluginlib/CMakeLists.txt

```diff
@@ -30,47 +30,96 @@ install(
   DESTINATION include
 )
 
-# if(BUILD_TESTING)
-#   add_library(test_plugins EXCLUDE_FROM_ALL SHARED test/test_plugins.cpp)
-#   target_include_directories(test_plugins PUBLIC
-#     include
-#     ${ament_index_cpp_INCLUDE_DIRS}
-#     ${class_loader_INCLUDE_DIRS}
-#     ${tinyxml2_INCLUDE_DIRS}
-#   )
-#   # place the library in an FHS compliant layout in the build space
-#   set(prefix "${CMAKE_CURRENT_BINARY_DIR}/prefix")
-#   set_target_properties(
-#     test_plugins PROPERTIES
-#     ARCHIVE_OUTPUT_DIRECTORY "${prefix}/lib"
-#     LIBRARY_OUTPUT_DIRECTORY "${prefix}/lib"
-#     RUNTIME_OUTPUT_DIRECTORY "${prefix}/bin"
-#   )
-#   # place the plugin manifest in an FHS compliant layout in the build space
-#   file(COPY test/test_plugins.xml DESTINATION "${prefix}/share/pluginlib/test")
-#   # create an entry in the ament index for the exported plugins
-#   file(WRITE "${prefix}/share/ament_index/resource_index/pluginlib__plugin/pluginlib"
-#     "\${prefix}/test/test_plugins.xml\n")
-
-#   catkin_add_gtest(${PROJECT_NAME}_utest test/utest.cpp)
-#   if(TARGET ${PROJECT_NAME}_utest)
-#     target_link_libraries(${PROJECT_NAME}_utest ${tinyxml2_LIBRARIES} ${catkin_LIBRARIES})
-#     add_dependencies(${PROJECT_NAME}_utest test_plugins)
-#   endif()
-
-#   include(CheckCXXCompilerFlag)
-#   check_cxx_compiler_flag("-std=c++11" COMPILER_SUPPORTS_CXX11)
-#   if(COMPILER_SUPPORTS_CXX11)
-#     catkin_add_gtest(${PROJECT_NAME}_unique_ptr_test test/unique_ptr_test.cpp)
-#     if(TARGET ${PROJECT_NAME}_unique_ptr_test)
-#       target_link_libraries(${PROJECT_NAME}_unique_ptr_test
-#         ${tinyxml2_LIBRARIES} ${catkin_LIBRARIES})
-#       set_target_properties(${PROJECT_NAME}_unique_ptr_test
-#         PROPERTIES COMPILE_FLAGS -std=c++11 LINK_FLAGS -std=c++11)
-#       add_dependencies(${PROJECT_NAME}_unique_ptr_test test_plugins)
-#     endif()
-#   endif()
-# endif()
+
+if(BUILD_TESTING)
+  find_package(ament_cmake_gtest REQUIRED)
+
+  set(prefix "${CMAKE_CURRENT_BINARY_DIR}/prefix")
+
+  # Write the necessary files for ament to detect a package.
+  file(COPY test/test_plugins.xml DESTINATION "${prefix}/share/pluginlib/")
+  file(COPY test/package.xml DESTINATION "${prefix}/share/pluginlib/")
+  file(WRITE "${prefix}/share/ament_index/resource_index/pluginlib__pluginlib__plugin/pluginlib"
+    "share/pluginlib/test_plugins.xml\n")
+  file(WRITE "${prefix}/share/ament_index/resource_index/packages/pluginlib" "")
+
+  add_library(test_plugins SHARED ./test/test_plugins.cpp)
+  target_link_libraries(test_plugins ${class_loader_LIBRARIES})
+  target_compile_definitions(test_plugins PUBLIC "PLUGINLIB__DISABLE_BOOST_FUNCTIONS")
+  target_compile_definitions(test_plugins PRIVATE "TEST_PLUGINLIB_FIXTURE_BUILDING_LIBRARY")
+  target_include_directories(test_plugins
+      PUBLIC
+      include
+      test/include
+      ${ament_index_cpp_INCLUDE_DIRS}
+      ${class_loader_INCLUDE_DIRS}
+      ${rcutils_INCLUDE_DIRS}
+      ${TinyXML2_INCLUDE_DIRS}
```

Comment:
For ROS 2 packages you could use `ament_target_dependencies` to add include directories, libraries etc. to each target.